### PR TITLE
Verify that user can apply the filter for each category within the Category dd list and reset the filter

### PR DESCRIPTION
**Pull request by gudzenkoleruska**
https://trello.com/c/DE6E29t4/470-tc-151103-verify-that-user-can-apply-the-filter-for-each-category-within-the-category-dd-list

### CODE REVIEW

File: tests/menTops.spec.js [hunk 1]
```diff
@@ -108,4 +108,41 @@ test.describe('menTops', () => {
...
+    'Jackets',
+    'Hoodies & Sweatshirts',
+    'Tees',
+    'Tanks'
+  ]
+
+  const topMenPage = 'men/tops-men.html';
+
+  for (let i = 0; i < categoriesList.length; i++) {
+    await page.goto(topMenPage);
```
Comment (SvetlanaKanel) [inline tests/menTops.spec.js:137]: You cannot start from the topMenPage. You need to add steps on how the user comes to this page starting from the homePage.

Comment (gudzenkoleruska) [reply tests/menTops.spec.js:137]: @SvetlanaKanel thanks for you comment, but in my preconditions already mentioned that user is located in this specific page and not need to do additional steps from Home page as it is checked within another tests.

**Review by OlgaPerepelitzki — DISMISSED** [review]

**Review by IrenaKh95 — DISMISSED** [review]

File: tests/menTops.spec.js (hunk 1 above)
Comment (MariaMoroz) [reply tests/menTops.spec.js:147]: @gudzenkoleruska Комментарий Светланы вполне резонный. Оформляйте ваш precondition простыми шагами - кликами по ссылкам. await page.goto(topMenPage); - не должно быть в тесте

Comment (gudzenkoleruska) [reply tests/menTops.spec.js]: added the preconditions steps to test

Comment (BabintsevaS) [reply tests/menTops.spec.js]: @gudzenkoleruska на 148 строке -  await page.goto('men/tops-men.html'); вы же все равно переходите напрямую на страницу, а нужно идти так как бы это делал пользователь, начиная с homepage.

File: tests/menTops.spec.js [hunk 2]
```diff
@@ -108,11 +110,58 @@ test.describe('menTops', () => {
...
  });
+
+
+
+
+
+
+
+
+
```
Comment (S-Lanchik) [inline tests/menTops.spec.js:121]: Why did you add empty lines?

File: tests/menTops.spec.js [hunk 3]
```diff
@@ -108,11 +110,58 @@ test.describe('menTops', () => {
...
+    'Hoodies & Sweatshirts',
+    'Tees',
+    'Tanks'
+  ]
+
+  await page.locator('#ui-id-5').hover();
+  await page.locator('#ui-id-17').click();
+
+  for (let i = 0; i < categoriesList.length; i++) {
+    await page.goto('men/tops-men.html');
```
Comment (S-Lanchik) [inline tests/menTops.spec.js:148]: If i understand the test correctly, you do not need this line since your steps on lines 144-145 redirect to this page.

File: tests/menTops.spec.js (hunk 2 above)
Comment (gudzenkoleruska) [reply tests/menTops.spec.js]: deleted empty rows

File: tests/menTops.spec.js (hunk 3 above)
Comment (gudzenkoleruska) [reply tests/menTops.spec.js]: deleted the line:  await page.goto('men/tops-men.html');

File: tests/menTops.spec.js (hunk 1 above)
Comment (gudzenkoleruska) [reply tests/menTops.spec.js]: @BabintsevaS deleted from the test:  await page.goto('men/tops-men.html')

File: tests/menTops.spec.js [hunk 4]
```diff
@@ -108,11 +110,48 @@ test.describe('menTops', () => {
...
+ test('Verify that user can apply the filter for each category within the Category dd list and reset the filter', async ({page}) =>{
+  const categoriesList = [
+    '.filter-options-item.allow.active > div.filter-options-content > ol > li:nth-child(1) > a',
+    '.filter-options-item.allow.active > div.filter-options-content > ol > li:nth-child(2) > a',
+    '.filter-options-item.allow.active > div.filter-options-content > ol > li:nth-child(3) > a',
+    '.filter-options-item.allow.active > div.filter-options-content > ol > li:nth-child(4) > a'
+  ]
+
+  const subcategoryLinks = [
+    'https://magento.softwaretestingboard.com/men/tops-men.html?cat=14',
```
Comment (verafes) [inline tests/menTops.spec.js:123]: '/' is our base URL defined in our project settings: and "/" = "https://magento.softwaretestingboard.com". So, these links could be like this: '/' + '/men/tops-men.html?cat=14'

**Review by verafes — APPROVED** [review]

**Review by mlimau — APPROVED** [review]
